pymongo/hello.py

```diff
@@ -12,11 +12,192 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-"""Helpers for the 'hello' and legacy hello commands."""
+"""Parse a response to the 'hello' or legacy hello command."""
+import itertools
+
+from pymongo import common
+from pymongo.server_type import SERVER_TYPE
 
 
 class HelloCompat:
     CMD = 'hello'
     LEGACY_CMD = 'ismaster'
     PRIMARY = 'isWritablePrimary'
     LEGACY_PRIMARY = 'ismaster'
+
+
+class Hello(object):
+    __slots__ = ('_doc', '_server_type', '_is_writable', '_is_readable',
+                 '_awaitable')
+
+    def __init__(self, doc, awaitable=False):
+        """Parse a hello response from the server.
+
+        .. versionadded:: 4.0
```

Comment:
This should say 3.12. Is this not a copy/paste from v3.12?

Reply:
It wasn't a copy-paste. I actually did a refactor. However, even what is in v3.12 doesn't have a `..versionadded` in the docstring.

I have updated the PR to use what is 3.12 verbatim as much as possible.